### PR TITLE
fix: Improve retry mechanisms for split pdf logic

### PR DESCRIPTION
**Pull request by jordan-homan**

### CODE REVIEW

File: src/hooks/custom/SplitPdfHook.ts
[the remark this answers is not in the document]
Comment (awalker4) [reply src/hooks/custom/SplitPdfHook.ts:1]: Good catch! Thanks

**Review by awalker4 — APPROVED** [review]

**Comment by awalker4** [comment]
Ah, one sec, we may need to hardcode our retry values in here - let me look.

File: src/hooks/custom/SplitPdfHook.ts [hunk 1]
```diff
@@ -68,9 +68,16 @@ export class SplitPdfHook
    * @returns The initialized SDK options.
    */
   sdkInit(opts: SDKInitOptions): SDKInitOptions {
-    const { baseURL, client } = opts;
-    this.client = client;
-    return { baseURL: baseURL, client: client };
+    const { baseURL } = opts;
+    this.client = new HTTPClientExtension();
+
+    this.client.addHook("response", (res) => {
```
Comment (jordan-homan) [inline src/hooks/custom/SplitPdfHook.ts:74]: maybe we could move this into the HTTPClientExtension class? not necessary though

Comment (awalker4) [reply src/hooks/custom/SplitPdfHook.ts:74]: Makes sense to me!

**Pull request merged**